File: lab3_1.kt
import kotlin.random.Random

class Money {
    private var amount = Random.nextInt(0, 100)

    init {
        println("Instantiated Money object with an amount of: $amount")
    }

    fun getAmount(): Int {
        return amount
    }
}

fun over50(objects: Array<Money>) {
    for (i in 0..4) {
        val tempAmount = objects[i].getAmount()
        if (tempAmount > 50) {
            println("amount of index $i and of value $tempAmount is greater than 50")
        }
    }
}

fun main() {
    val moneyArray = Array(5) { Money() }

    over50(moneyArray)
}

File: labs/lab3_1.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

class Money
{
private:
    int amount;

public:
    void set_amount()
    {
        amount = rand() % 100 + 1;
    }

    int get_amount()
    {
        return amount;
    }
};

int main()
{

    Money *money_ptr = new Money[3];

    for (int i = 0; i < 5; i++)
    {
        money_ptr[i].set_amount();
        cout << "amount of money " << money_ptr[i].get_amount() << "\n";
    }

    for (int i = 0; i < 5; i++)
    {
        if (money_ptr[i].get_amount() > 50)
        {
            cout << "amount of index " << i << " and of value " << money_ptr[i].get_amount() << " is greater than 50 \n ";
        }
    }

    return 0;
}